paddle/operators/box_coder_op.h

```diff
@@ -0,0 +1,163 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+#include "paddle/framework/op_registry.h"
+#include "paddle/operators/math/math_function.h"
+
+namespace paddle {
+namespace operators {
+
+using Tensor = framework::Tensor;
+using LoDTensor = framework::LoDTensor;
+
+enum class BoxCodeType { kEncodeCenterSize = 0, kDecodeCenterSize = 1 };
+
+inline BoxCodeType GetBoxCodeType(const std::string& type) {
+  if (type == "encode_center_size") {
+    return BoxCodeType::kEncodeCenterSize;
+  } else if (type == "decode_center_size") {
+    return BoxCodeType::kDecodeCenterSize;
+  }
+  PADDLE_THROW("Not support type %s.", type);
+}
+
+template <typename T>
+class BoxCoderKernel : public framework::OpKernel<T> {
+ public:
+  void EncodeCenterSize(const Tensor& target_box, const Tensor& prior_box,
+                        const Tensor& prior_box_var, T* output) const {
+    PADDLE_ENFORCE_EQ(target_box.dims().size(), 2,
+                      "The rank of target_box must be 2.");
+    PADDLE_ENFORCE_EQ(prior_box.dims().size(), 2,
+                      "The rank of prior_box must be 2.");
+    PADDLE_ENFORCE_EQ(prior_box_var.dims().size(), 2,
+                      "The rank of prior_box_var must be 2.");
+    PADDLE_ENFORCE_EQ(prior_box.dims()[0], prior_box_var.dims()[0],
+                      "The dims of prior_box must equal to prior_box_var.");
```

Comment:
All these checks have been done in the `InferShape`, can be removed here.

Reply:
done